onadata/apps/logger/tests/test_digest_authentication.py

```diff
@@ -60,10 +61,9 @@ def test_fail_authenticated_submissions_to_wrong_account(self):
         # Authentication required
         self.assertEqual(self.response.status_code, 401)
         auth = DigestAuth('dennis', 'dennis')
-        self._make_submission(xml_submission_file_path, add_uuid=True,
-                              auth=auth)
-        # Not allowed
-        self.assertEqual(self.response.status_code, 403)
+        with self.assertRaises(Http404):
+            self._make_submission(xml_submission_file_path, add_uuid=True,
+                                  auth=auth)
```

Comment:
Yes, this fails because they didn't provide the `username` of the user who owns the username and because we couldn't find a form matching the submission within the request users account.